api/Service/EventService.cs

```diff
@@ -137,5 +137,27 @@ public async Task<IList<Event>> GetFutureEventList()
 				 .Where(x => (x.StartingDate >= DateTime.Now)).ToListAsync();
 			return events;
 		}
+		public async Task<IList<Event>> GetFilteredEvents(FiltersDto filtersDto)
+		{
+			var eventsList = from Events in _context.Events select Events;
+
+
+			if (filtersDto.Category != null)
+			{
+				var categoriesMatching = from category in _context.Categories where (category.Title == filtersDto.Category) select category.Id;
+				eventsList = eventsList.Where(b => categoriesMatching.Contains(b.CategoryId));
+			}
+
+			if (filtersDto.EndingDate != null)
```

Comment:
When using nullable values, you should check if `filtersDto.EndingDate.HasValue`